internal/compiler/passes/const_propagation.rs

```diff
@@ -20,7 +20,9 @@ pub fn const_propagation(component: &Component) {
 fn simplify_expression(expr: &mut Expression) -> bool {
     match expr {
         Expression::PropertyReference(nr) => {
-            if nr.is_constant() {
+            if nr.is_constant()
+                && !matches!(nr.ty(), Type::Struct { name: Some(name), .. } if name.ends_with("::StateInfo"))
```

Comment:
Perhaps it's worth it to move this into a helper function - the same string check is three more places AFAICT.

Reply:
yeah, the other place are in the llr when we already wrote this information in a boolean.